src/Http/Request/Body.php

```diff
@@ -46,15 +46,11 @@ public function __construct(array|string|null $contents = null)
 	 */
 	public function contents(): string|array
 	{
-		if ($this->contents === null) {
-			if (empty($_POST) === false) {
-				$this->contents = $_POST;
-			} else {
-				$this->contents = file_get_contents('php://input');
-			}
+		if (empty($_POST) === false) {
+			return $this->contents ??= $_POST;
 		}
 
-		return $this->contents;
+		return $this->contents ??= file_get_contents('php://input');
```

Comment:
IMHO, although the changes are short and correct, I think the readability of the code here is very low. I think we could simply write it like following. Maybe longer, but it's easier to understand that it has to look in the cache first.

````php
public function contents(): string|array
{
    if ($this->contents !== null) {
        return $this->contents;
    }

    if (empty($_POST) === false) {
        return $this->contents = $_POST;
    }

    return $this->contents = file_get_contents('php://input');
}
````

Reply:
Or seeing `??=` as one and on a single line also looks good for readability.

````php
public function contents(): string|array
{
    return $this->contents ??= empty($_POST) === false ? $_POST : file_get_contents('php://input');
}

Reply:
I entirely agree with Ahmet and really like his first suggestion.

It's so much easier to read as the different logic alternatives have their own block, so mentally you can think about each alternative separately and don't need to consider all the ways how the logic could intertwine.

Also it makes testing much easier as we immediately know there are three cases to test (which can be confirmed with coverage).

Even though the PHP 8 syntax is really nice and useful in many places, I think we need to be super careful not to use it in places where it may even be detrimental and could cause hard to debug bugs in the future – just for a few saved lines that don't affect our users at all.

Reply:
You are right 👍